### PR TITLE
Feature/#438 update grid

### LINKED ISSUE #438: Audit Grid
We've had several iterations of our grid with/without Neat. We still utilize a grid, or grid classes, for styling our theme. However, I'm not sure these are even used and/or are ignored when we get into scaffolding. I say this mostly because styles for the `template-sidebar-right.php` utilizes styles inside `_layout.scss` and not our presentational classes.
- https://github.com/WebDevStudios/wd_s/blob/master/template-sidebar-right.php#L14
- https://github.com/WebDevStudios/wd_s/blob/master/assets/sass/structure/_layout.scss#L32
- https://github.com/WebDevStudios/wd_s/blob/master/assets/sass/structure/_layout.scss#L57

Another example: Our ACF recent posts block, which is a grid, utilizes additional styles from neat does not use our grid.
- https://github.com/WebDevStudios/wd_s/blob/master/assets/sass/modules/content-blocks/_recent-posts.scss.

It might be good to either get on the same page with the grid and document or remove in favor of user-styling.

### PR DESCRIPTION
**Pull request by jomurgel**
Closes #438

### DESCRIPTION ###
Much of the documentation for these updates were added to the documentation doc below: https://docs.google.com/document/d/1o1zfqeT5bcdu4VsCHCOKCbO72YMKjb7509I83srkzKU/edit#heading=h.b1ma2w6tlbd

**The main push here was removing Neat entirely and rebuilding the grid entirely with flexbox from top to bottom.**

**These grid helpers and mixins showed in the doc above removes the unnecessary flexibility of Neat 1.9x and provides the ability to create the following grid/layout options.**
- **Full Width**
- **Container Width**
- **50%**
- **33%**
- **25%**
- **~66% — Less Gutter accommodation.**

### Updates
- Added .vscode to .gitignore.
- Replaces any instance of the commonly-used rem(24) with a new $space variable.
- Added general style tweaks for IE11 or general accommodations for the new grid.
- Button styles to be more cross-browser compatible.
- Block naming conventions (below).
- Updated scaffolding to use any of the added mixins or classnames.
- Cleaned up hamburger open/close button.
- Fixed search form container elements.
- Added function 
- Replace normalize with Sanitize — seen in #442 
- Inlines Instances of Triangle Mixin with inline styles.

screenshot 2018-12-17 11 18 36

### Removes
- Bourbon as a dependency
- Bourbon from Gulpfile.js
- Hyphens Mixin
- Font Legibility Mixin
- Margin Auto Mixin
- Margin Padding Reset Mixin
- Vertical Align Mixin
- Word Break Mixin
- Z-Index Mixin
- Need for Clearfix Mixin, removed mixin, and styled the areas that used it to use flexbox.

### Added
- New template-sidebar-left.php template.
- Empty style partial for the generic content block.
- Default styles for blockquote to match typical output (extends h3).
- Added function to push YOAST below all content to hooks.php
- Added function to escape content output the way that WP does it — replacing any instances of force_balance_tags() which doesn’t escape anything.
- Directional Property Mixin — helper.
- Margin Mixin
- Padding Mixin
- Position Mixin
- Shorthand Function

I made several opinionated additions here that we're constantly making updates during projects. Those are listed here: https://docs.google.com/document/d/1o1zfqeT5bcdu4VsCHCOKCbO72YMKjb7509I83srkzKU/edit#heading=h.b1ma2w6tlbd

I still think there are many areas we could improve the mixin usage and markup, but there are other issues for that.

### SCREENSHOTS ###
**ACF Blocks**
screencapture-192-168-1-237-3000-acf-blocks-2018-12-17-11_10_57

**Quarter Grid**
screenshot 2018-12-17 11 13 35

**Radio/Checkbox/Select Updates**
screen shot 2018-12-17 at 10 54 25 am

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT?

### DOCUMENTATION ###
The doc below has updates and documentation for these changes and would, if approved, be added to the Wiki: 
https://docs.google.com/document/d/1o1zfqeT5bcdu4VsCHCOKCbO72YMKjb7509I83srkzKU/edit#

### CODE REVIEW

File: 404.php
Comment (coreymcollins) [inline 404.php]: Since we're not escaping the apostrophe in `can't` here, I'm guessing it's rich text and should be replaced & escaped to match the heading on line 16.

File: archive.php
Comment (coreymcollins) [inline archive.php]: `role` is invalid here and shouldn't be used: https://www.w3.org/TR/html-aria/#document-conformance-requirements-for-use-of-aria-attributes-in-html

File: assets/sass/base/_scaffolding.scss
Comment (coreymcollins) [inline assets/sass/base/_scaffolding.scss]: We don't use the `primary` class at all anymore with this update, do we?

File: assets/sass/base/_typography.scss
Comment (coreymcollins) [inline assets/sass/base/_typography.scss]: Can we also add `.h#{$i}` here so we can capture not only the actual headings but the heading classes like `.h1`, `.h2`, etc?

File: assets/sass/modules/content-blocks/_fifty-fifty.scss
Comment (coreymcollins) [inline assets/sass/modules/content-blocks/_fifty-fifty.scss]: If this is empty now, let's just remove the partial. It can always be added if we need to adjust styles, but I'd rather avoid having an empty file just for the sake of having it.

File: assets/sass/modules/content-blocks/_general.scss
Comment (coreymcollins) [inline assets/sass/modules/content-blocks/_general.scss]: Same note as on fifty-fifty – if this is empty, let's just remove it.

File: assets/sass/modules/content-blocks/_global.scss
Comment (coreymcollins) [inline assets/sass/modules/content-blocks/_global.scss:17]: I think this should also include left and right padding. If something is container-width and has a background set, the content runs right up to the edge.

File: assets/sass/modules/content-blocks/_recent-posts.scss
Comment (coreymcollins) [inline assets/sass/modules/content-blocks/_recent-posts.scss]: If empty, let's remove.

File: assets/sass/modules/content-blocks/_related-posts.scss
Comment (coreymcollins) [inline assets/sass/modules/content-blocks/_related-posts.scss]: If empty, let's remove.

File: assets/sass/modules/navigation/_post-navigation.scss
Comment (coreymcollins) [inline assets/sass/modules/navigation/_post-navigation.scss]: I'd like to avoid using `@extend` where possible as it can cause headaches. Is there any reason to not add `display-flex` as a class to this element?

File: assets/sass/modules/navigation/_post-navigation.scss
Comment (coreymcollins) [inline assets/sass/modules/navigation/_post-navigation.scss]: Let's remove empty blocks.

File: assets/sass/plugins/_gravity-forms.scss
Comment (coreymcollins) [inline assets/sass/plugins/_gravity-forms.scss:144]: Do we need to do the math here? Could we just write `6px` instead since we're not doing math on a variable or anything like that?

File: assets/sass/structure/_layout.scss
Comment (coreymcollins) [inline assets/sass/structure/_layout.scss]: Remove extra space after `auto`

File: assets/sass/utilities/mixins/_pxtoem.scss
Comment (coreymcollins) [inline assets/sass/utilities/mixins/_pxtoem.scss]: Can you add an example like you have in the other mixin files?

File: assets/sass/utilities/mixins/_pxtorem.scss
Comment (coreymcollins) [inline assets/sass/utilities/mixins/_pxtorem.scss]: Can you add an example like you have in the other mixin files?

File: buddypress.php
Comment (coreymcollins) [inline buddypress.php]: Remove role.

File: inc/template-tags.php
Comment (coreymcollins) [inline inc/template-tags.php]: We should add aria labels here to specify what the nav is (something like `aria-label="<?php esc_html_e( 'Numeric pagination', 'swe' ); ?>"`.

File: index.php
Comment (coreymcollins) [inline index.php]: Remove role.

File: page.php
Comment (coreymcollins) [inline page.php]: Remove role.

File: search.php
Comment (coreymcollins) [inline search.php]: Remove role.

File: single.php
Comment (coreymcollins) [inline single.php]: Remove role.

File: template-acf.php
Comment (coreymcollins) [inline template-acf.php]: Remove role.

File: template-parts/content-blocks/block-accordion.php
Comment (coreymcollins) [inline template-parts/content-blocks/block-accordion.php:21]: There should be a fallback if no animation class is set, otherwise we'll get an empty `class` attribute which will throw a11y warnings.

File: template-parts/content-blocks/block-accordion.php
Comment (coreymcollins) [inline template-parts/content-blocks/block-accordion.php]: Should we run this through the new `s_get_the_content` template tag instead for consistency?

File: template-parts/content-blocks/block-accordion.php
Comment (coreymcollins) [inline template-parts/content-blocks/block-accordion.php:17]: Based on conversations we've had around section and div usage, should we make this a div instead?

File: template-parts/content-blocks/block-related_posts.php
Comment (coreymcollins) [inline template-parts/content-blocks/block-related_posts.php:31]: If no animation class is present, we'll wind up with an empty space here which will throw an a11y warning.

File: template-parts/content-password-protected.php
Comment (coreymcollins) [inline template-parts/content-password-protected.php]: Let's remove the empty class attribute.

File: template-scaffolding.php
Comment (coreymcollins) [inline template-scaffolding.php]: Remove role.

File: template-sidebar-left.php
Comment (coreymcollins) [inline template-sidebar-left.php]: Remove role.

File: template-sidebar-right.php
Comment (coreymcollins) [inline template-sidebar-right.php]: Remove role.

**Review by coreymcollins — CHANGES_REQUESTED** [review]
I've run through and reviewed the actual code changes, leaving a handful of comments here and there.

Before you get started on anything, I'm also going to run some visual QA tests (and try to break things!) as well as linting tests, so I'll add those as additional comments once I'm wrapped up with those.

**Comment by coreymcollins** [comment]
I’m leaving visual notes on the theme in general. Some of these items are going to be due to grid changes and some may be issues present already. In the case of the latter, fixing those issues as part of this PR would be super helpful!

# Visual Notes

## Header

The search form no longer extends fully when opened

## 404

From mid-size down, the content runs into the edge of the screen. This may be a global issue but this is the first page I’m checking, so maybe not! Either way, we’ll want to make sure pages have left and right margin/padding to avoid the collision.

## Grid

When sizing down the window, the content runs into the edge of the screen until we hit the 1170px breakpoint where the container gets left and right padding. You can see some edge collisions in the header as well.

## ACF Blocks

When a block has a background and is `container-width`, it needs some padding so the content doesn't hit the edge. This is not an issue below 1170px.

## Mobile Nav

The link color is hard to read on the black background.

## Comments Mobile

When a lot of comments are replies, they smush up real good on small screens.



The comment section and form run into the edges of the screen on mobile.

## Buttons


Button hover/active states are super hard to read.

# Accessibility

## WCAG & 508
[a11y.zip](https://github.com/WebDevStudios/wd_s/files/2765948/a11y.zip)
Files attached here with some issues. Some of these are already noted in the code review, so just crosscheck to make sure we’re not missing anything.

I ran this on just the ACF Blocks page for now. 
I’d like to run this again once we get `master` merged back in as there have been a bunch of additions to the Scaffolding template which would be good to examine.

# CBT
Be sure to run CBT on main pages – the ACF Blocks page, the Scaffolding page (after merging in master), single posts, etc. Basically, let’s make sure any unique page is looking good.

# Merge `master`
Before passing this back for review, can you merge `master` into your branch? We’ve merged in the Sanitize and Scaffolding PRs and there are some conflicts that should be cleaned up + seeing how the new scaffolding elements look with your changes.

Thanks for diving so hard into this! This is looking really great, and creating a grid is pretty dang easy. I think one thing we’ll want to keep in mind is to communicate with our designers (i.e., you and @bezierer) to make sure we’re providing mockups that will be in line with our grid at whatever max content width we run at for each project. I wouldn’t want to provide mockups for a client with a grid that looks one way and then have to spend time customizing the grid in the theme to match super specific sizes if we can avoid that by keeping the mockups and CSS as closely tied to one another as possible.

File: archive.php
Comment (jomurgel) [reply archive.php]: Ah, thanks. I thought that was a thing. I know the older WP core themes doubled up that way so I erred on the side of caution.

File: assets/sass/base/_scaffolding.scss
Comment (jomurgel) [reply assets/sass/base/_scaffolding.scss]: Ah, no. This should have been removed.

File: assets/sass/plugins/_gravity-forms.scss
Comment (jomurgel) [reply assets/sass/plugins/_gravity-forms.scss:144]: @coreymcollins I haven't touched any of the gravity form styles other than moving things around and updating some variables. I guess I'm not sure what reason this was done and/or if it's a requirement. Making a note here before making the update.

File: template-parts/content-blocks/block-accordion.php
Comment (jomurgel) [reply template-parts/content-blocks/block-accordion.php:21]: Opted to set a default class inside the function rather than rewriting output in all of the template parts

**Review by ImBigWill — COMMENTED** [review]
Rather than a screenshot, I just mentioned a small cleanup item similar to the 404 note, but for the content-none.php escaping of text. Overall this looks really good and I agree with Corey probably need to run through it after some of the updates since there are quite a few.

File: template-parts/content-blocks/block-accordion.php
Comment (jomurgel) [reply template-parts/content-blocks/block-accordion.php:17]: @coreymcollins we might want to discuss this further. I think this would sort of justfiy having some function + acf field to determine the output. I could argue for div or even an aside or article depending on the content.

Comment (jomurgel) [reply template-parts/content-blocks/block-accordion.php:17]: The only one, based on intended use, that would for sure be a div or aside would be the cta block.

Comment (coreymcollins) [reply template-parts/content-blocks/block-accordion.php:17]: I figured this would be one we could all chat about, so I added an agenda item for today's scrum. I'm sure we'll all come to a decision on what would be best for us + users.

**Comment by jomurgel** [comment]
@coreymcollins a few followup questions.

Can you double check the search form again for me with some changes I've pushed up. This is what I see, but I also don't know if there's some part that is missing or some particular page that might be happening on, etc.

screenshot 2019-01-17 14 19 37

- Updated the 404 page
- Accommodated padding with the container class with an offset helper, breaking ~48px before the container to accommodate the padding offset.
- Updated the mobile menu styles to work better with the default WD_s colors.

screenshot 2019-01-17 14 42 49

- Decreased padding on replies to help out mobile a bit.

screenshot 2019-01-17 15 02 46

- Fixes container wrapping in comments prevents butting up against the edges.
- Resolves issue with button text readability on hover/focus.

screenshot 2019-01-17 15 08 13

Several other visual items were addressed during code note updates.

Need to run through WCAG and 508 elements and CBT stuff yet. Master was merged back in, however, on this round of updates. Need to address mentioned `@extend` elements in the styles as well.

**Comment by jomurgel** [comment]
I've removed, in the post navigation, the use of `@extend` by duplicating the styles from `_layout.scss`. There's no filter to update `.posts-navigation` or `.post-navigation` so this was the cleanest way of doing it. Though, for the record, I don't think this is the best way to go about it either. I think using `@extend` here would have been perfectly fine.

Going through the WCAG and 508 errors now, though it appears that most of these are resolved at a glance.

Will make updates and hopefully be able to run CBT here shortly.

**Comment by jomurgel** [comment]
@coreymcollins I think I resolved the WCAG and 508 errors locally. Might try pulling down the branch and re-running your reports to confirm.

The only one I couldn't figure out what the `slick.css` being enqueued in the footer. I can't figure out why that's happening and/or how to move it if you have any ideas. You'd mentioned it potentially being done on purpose for Pagespeed Insights or some render-blocking resources issue, but I couldn't not see any reason that it would be enqueued in a place outside of the norm.

Went through and fixed a few warnings also, but most of those are non-issues, or were resolved by completely including image alt text in the admin or values where missing.

Ignored:
- [WCAG2 2.2.2 (A)] Videos that auto-play must have a means to stop them. (background videos)
- Whitespace warnings
- This warning
screenshot 2019-01-18 09 54 21

Which must be an issue from older code since the nav has an aria label: https://github.com/WebDevStudios/wd_s/blob/feature/%23438-update-grid/inc/template-tags.php#L535

I ran CBT here, which is a little weird running from a local, but as far as I can tell things, in reality, look fine. 
- https://app.crossbrowsertesting.com/public/i55b132d134e1a40/screenshots/z4c8daa3e83745f97613

iPad and iPhone didn't work. I couldn't even get the live tests to work. Checking in the mobile chrome checker looks fine, those are here, but might need to find someone able to test an actual new device locally.
iPhone
screencapture-192-168-1-237-3000-acf-blocks-2019-01-18-10_26_17

iPad
screencapture-192-168-1-237-3000-acf-blocks-2019-01-18-10_26_03

**Comment by coreymcollins** [comment]
Just digging into this now and the first thing I wanted to test was the Slick Carousel stuff.

Right now we're enqueuing the scripts & styles in the template part for the carousel block. This is what puts the scripts and styles in the footer. If I enqueue these elsewhere, like in `scripts.php` they show up in the header.

Going to look at, instead of enqueuing the scripts directly in the template part, calling a function that enqueues them at the proper time and place. I'll update with findings!

**Comment by coreymcollins** [comment]
This commit (https://github.com/WebDevStudios/wd_s/pull/447/commits/610ff58cedac3d86886b8a904325e36ee0f90c25) takes care of the issue of Slick CSS being enqueued in the footer.

JS still gets enqueued in the footer, as expected, but the CSS does not. We were calling `wp_enqueue_style` and `wp_enqueue_script` directly in the carousel content block template part which is what was causing the issue. I moved those enqueues to their own function (`_s_enqueue_slick_scripts`) and am calling that function in the carousel template part instead.

**Comment by coreymcollins** [comment]
The search in the header is still causing an issue for me and it's kind of weird. Up until 1800px, I can see it expanding:


Once I get past 1800px, it becomes the crunched up little dude seen here:


Still running through the rest of the updates but wanted to log that one first since you had asked about it specifically.

**Comment by coreymcollins** [comment]
I pushed up a handful of commits to take care of some a11y things, including adding a play/pause button to our blocks when a background video is used. This allows users who may want or need to pause a background video to do so. TVP still throws a warning around this as we're not using the native video controls but a JS solution with a button, but it is accessible/tab-able and includes screen reader text for an added bonus.

With your note on the `<nav>` item – this looks to be thrown because the primary nav doesn't have an aria label set. This is because we're using the core `wp_nav_menu` function and I don't think there's a way to add aria labels. We may be able to filter the output or maybe add a `<nav>` in the markup with the labels we need and change the way we output the menu?

Aside from the items noted above, I am seeing one last screen collision item noted below. Otherwise, this is looking really great!

I know Will looked it over and was happy with it, and I'd love to see if any other FEEs may want to give it a test to catch anything we may have missed (if anything) early next week. I don't want to delay it too long, though, as this is a big and good update that will be awesome to start using.

### Screen Collisions
The edge-of-screen collisions are looking pretty good for the most part, but the nav in the header still runs into the edge around 1170px as seen here:


It looks like around 1194px that things jump back into place:

**Comment by jomurgel** [comment]
Pushed up another commit that adds the aria-label to the main navigation mentioned above.

``` php
		<nav id="site-navigation" class="main-navigation" aria-label="<?php esc_html_e( 'Main Navigation', '_s' ); ?>">
			<?php
			wp_nav_menu( array(
				'fallback_cb'     => false,
				'theme_location'  => 'primary',
				'menu_id'         => 'primary-menu',
				'menu_class'      => 'menu dropdown',
				'container'       => false,
			) );
			?>
		</nav><!-- #site-navigation-->
```

**Comment by jomurgel** [comment]
Pushed a couple additional updates to fix the nav issue above, and brought the branch current with master. 👍

**Comment by coreymcollins** [comment]
@jomurgel is this ready for review again? I see your updates above but it's still assigned to you with the `returned` label, so I wasn't sure where it stood at this point. Thanks!

**Comment by jomurgel** [comment]
@coreymcollins updated a couple things here:
- Resolved issues with the width of the search form shown in your screenshots above. Removed any real width restriction above desktop devices.
- Also resolved an issue with the use of `calc` and variables in the grid mixins which was just breaking the value of `flex-basis`.

**Review by gregrickaby — CHANGES_REQUESTED** [review]
Hey @jomurgel and @coreymcollins! Great work here. I really appreciate dropping the sidebar, and all other the little details. Here are a couple of suggestions:

1. Move _Font Color_ and _Animation_ from "Other Options" --> "Display Options".
2. When I set a block to Full [width] in Display Options, the background doesn't actually extend full width. It could be, that I misunderstand what full width should be?

**Comment by coreymcollins** [comment]
Thanks, Greg! Interesting catch on no. 2 above. That definitely was extending full-width previously (per the screenshots earlier in the PR), so there must be a commit here that walked that back at some point.

I think the adjustment in no. 1 makes sense as well. 👍

**Comment by gregrickaby** [comment]
@coreymcollins I even re-ran Gulp just to make sure. It's not full-width:

screenshot

**Comment by coreymcollins** [comment]
Yup, looks like this commit is the culprit by adding `container` as a class: https://github.com/WebDevStudios/wd_s/pull/447/commits/482d0938b1e01b18e1d1556e3719d39f2c9646cd

I'll get with Jo to see if adding that class was intended or if we can remove it without issue.

**Review by coreymcollins — APPROVED** [review]

**Comment by coreymcollins** [comment]
Pushed some updates here to:
- Move Animation & Font Color to Display Options (https://github.com/WebDevStudios/wd_s/pull/447/commits/733fde739b4cb9903b59a9d51dbb4182001e64cc)
- Update the ACF functions to look for those options in their new spots (https://github.com/WebDevStudios/wd_s/pull/447/commits/733fde739b4cb9903b59a9d51dbb4182001e64cc)

- Update the carousel module to accept the animation and color options in its parent container (https://github.com/WebDevStudios/wd_s/pull/447/commits/196b2be2a6a9e705d52ef12abcb99ab0eb080575). Any attempt at trying to grab a gif of this failed miserably with a super choppy gif 😆 
- Fix a couple of space issues (https://github.com/WebDevStudios/wd_s/pull/447/commits/7465d707e30870e7608b523dff7a38f8f32feae9, https://github.com/WebDevStudios/wd_s/pull/447/commits/9a5937d5fa752a54c4e0578dee77487f803f9584)

**Comment by gregrickaby** [comment]
@coreymcollins Awesome! I think we can do it even better. Background Options are technically part of the front-end display, and should probably be consolidated to Display Options.

**Comment by coreymcollins** [comment]
I like this idea in theory, but one sticky thing is that we don't offer Background Options on the Carousel since each slide has its own options for setting the slide content & background.

If we move Background Options into Display Options, we run into a problem where they display in that tab but won't actually be used for the Carousel block. I don't think there's a way to conditionally hide a set of fields in a tab group based on the content block type unless we throw some JS in there.

Going to think through it some more and see if there's a way to move them and still elegantly hide them for blocks that won't use Background Options.

**Comment by coreymcollins** [comment]
I think the best call may be to keep Background Options as-is so we can keep them off of the Carousel block (and any other blocks that may not need/be able to use background options) and merge this puppy in if everything else is looking good from your review.

We can always explore sneaky ways to manipulate tabs based on flexible content layouts further, but I'd love to get the grid stuff updated and merged in today if at all possible. Let me know what you think when you have a chance!

**Review by gregrickaby — APPROVED** [review]

**Comment by gregrickaby** [comment]
@coreymcollins Sounds good. Thanks for taking a look!

**Pull request merged**
